### ISSUE TITLE
Android studio freezes on layout editor

### ISSUE THREAD

**Issue by adigyran** [stepstone-tech/android-material-stepper#202]
Version :4.1.0
Android studio 2.3.3

Layout editor freezes on any edit or visual rendering. Only on layouts xml with stepper layout.
Freezing is continuous, only restart helps 
4.0.0 version works fine.

**Comment by huttneab** [comment]
I'm seeing the same behavior.

**Comment by RobertZagorski** [comment]
Indeed, I can confirm that the following configuration causes the layout to freeze:

    <com.stepstone.stepper.StepperLayout
           android:id="@+id/stepperLayout"
           android:layout_width="match_parent"
           android:layout_height="wrap_content"
           app:ms_stepperType="tabs" />

The error is:
```
java.lang.InstantiationException: The layout involves creation of com.stepstone.stepper.internal.widget.StepTab over 100 levels deep. Infinite recursion?
       	at org.jetbrains.android.uipreview.ViewLoader.loadClass(ViewLoader.java:257)
       	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:220)
       	at com.android.tools.idea.rendering.LayoutlibCallbackImpl.loadView(LayoutlibCallbackImpl.java:186)
       	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:334)
       	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:345)
       	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:245)
       	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
       	at android.view.LayoutInflater.inflate(LayoutInflater.java:495)
       	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
       	at com.stepstone.stepper.internal.widget.StepTab.<init>(StepTab.java:125)
       	at com.stepstone.stepper.internal.widget.StepTab.<init>(StepTab.java:120)
       	at sun.reflect.GeneratedConstructorAccessor263.newInstance(Unknown Source)
       	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
       	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
       	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:475)
       	at org.jetbrains.android.uipreview.ViewLoader.loadClass(ViewLoader.java:262)
       	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:220)
       	at com.android.tools.idea.rendering.LayoutlibCallbackImpl.loadView(LayoutlibCallbackImpl.java:186)
       	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:334)
       	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:345)
       	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:245)
       	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
       	at android.view.LayoutInflater.inflate(LayoutInflater.java:495)
       	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
       	at com.stepstone.stepper.internal.widget.StepTab.<init>(StepTab.java:125)
       	at com.stepstone.stepper.internal.widget.StepTab.<init>(StepTab.java:120)
```

I would appreciate if you could confirm, whether this is your case or if there's another one?

**Comment by adigyran** [comment]
where to find it? In idea.log nothing like you pasted

**Comment by RobertZagorski** [comment]
Layout preview errors are shown at the bottom of preview. When dragging it up, details of error are shown.
However, the error is less important, I am more interested in how do you use `StepperLayout`. Is it the configuration I provided or a different one?

**Comment by adigyran** [comment]
you don`t get it, This problem freezes entire android studio, i can`t even do anything, only restart. I can`t drag anything 
```
 [<com.stepstone.stepper.StepperLayout
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:id="@+id/claims_stepper_layout"
            android:layout_below="@id/claims_steps_title"
            app:ms_stepperType="tabs"
            app:ms_tabNavigationEnabled="false"
            app:ms_inactiveStepColor="#ACACAC"
            app:ms_activeStepColor="@color/redTextLogin"
            app:ms_nextButtonText="@string/next"
            app:ms_showBottomNavigation="false"](url)
```
i`m using it like that

**Comment by Sirvanalie** [comment]
Hi. I'm seeing this problem, when I use layout for apply changeset , Android studio freezes . And even I can't close Android studio with Close button. Finally I closed by Task Manager 👎

**Comment by zawadz88** [comment]
Is the issue still reproducible in 4.2.0?
I'm using Android Studio 3.0.0-beta4 and I don't see this in the sample app - although the Preview does not work in general on this version for me even with material-stepper 2.3.2 (it was working on Android Studio 2.3.3)

**Comment by adigyran** [comment]
same android studio version, version 4.2.0 - no freezes. In preview stepper is just gray box, before that in layout preview was actual steps inidicators

**Comment by zawadz88** [comment]
This seems to be reported to Google: https://issuetracker.google.com/issues/65798740

**Comment by zawadz88** [comment]
Regarding
> In preview stepper is just gray box, before that in layout preview was actual steps inidicators

This seems like an Android Studio issue - I'll report this to Google.
As a workaround with Android Studio 3.0.0-beta6 and material-stepper 4.2.0 (in the sample app at least) you can add a theme attribute to the view directly, e.g:
```
<?xml version="1.0" encoding="utf-8"?>
<com.stepstone.stepper.StepperLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/stepperLayout"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    app:ms_stepperType="tabs"
    tools:theme="@style/Theme.AppCompat" />
```
Settings this via Select Theme in the Preview editor does not work for some reason...

Please let me know if this helps.

**Comment by zawadz88** [comment]
Issue created: https://issuetracker.google.com/issues/66902070

**Comment by zawadz88** [comment]
Hi @adigyran,
Are you experiencing the freezes in 4.3.0 as well?

**Comment by adigyran** [comment]
i will test that tomorrow.

**Comment by UlrichRaab** [comment]
Yes the freezes also happen when using 4.3.0 and Studio 2.3.3

**Comment by zawadz88** [comment]
Thanks, I can reproduce this.
It's a bit tricky as the sample project requires Android Studio 3.x but I'll try to find a workaround so that it does not freeze anymore.

**Comment by zawadz88** [comment]
This seems to be caused by using
```
    tools:parentTag="com.stepstone.stepper.internal.widget.StepTab"
```
in `layout/ms_step_tab.xml`

**Comment by zawadz88** [comment]
Fixed in 4.3.1.

As for the second issue in Android Studio 3.0 you can use the workaround described here.
The fix for that is to be in Android Studio 3.1: https://issuetracker.google.com/issues/66902070
